development/frameworks/ReactAppStructure.md

```diff
@@ -78,32 +106,73 @@ Here is an example of the file structure that should be used:
 │   │   └── index.js
 ```
 
-The component is in a folder of its name. It always has a `.js` file of it's name where the majority of the JavaScript is. It always has a `index.js` file to export the other js file. The component relevant `.scss` is here optionally, as well as an assets folder with any images etc. that the component needs.
+The component is in a folder of its name. It always has a `.js` file of it's name where the majority of the JavaScript is. It always has a `index.js` file to export the other `js` files. The component relevant `.scss` is here, as well as an assets folder with any images etc. that the component needs. Tests and `storybook` (if used) should also be in the folder.
 
 If you are using typescript, this is where your `.tsx` file will go.
 
-### Common tasks of a Component
-TBA
 
-## Containers
-Should store smart components which handle state for stateless components.
+#### Common tasks of a Component
+* Should handle any styling or display logic which is based on its props.
+* Should change or manipulate data by invoking a callback which is passed as a prop.
 
-### Naming convention
+### Containers
+Should store components which handle state for stateless components. They are either using `state` or connected to the `store`
```

Comment:
May be good for clarity to draw the distinction between app state and internal/or UI state